conans/test/functional/profiles/test_python_profiles.py

```diff
@@ -0,0 +1,164 @@
+import os
+import textwrap
+import unittest
+from collections import defaultdict
+
+from importlib.util import spec_from_file_location, module_from_spec
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+from conans import ProfileSettings, ProfileOptions
+
+
+class ProfileLoader(object):
+
+    @staticmethod
+    def _init_profile():
+        import conans
+        conans.settings = ProfileSettings()
+        conans.options = ProfileOptions()
+        conans.env = ProfileOptions()
+
+    @staticmethod
+    def _load_module_from_file(filename):
+        name = os.path.splitext(os.path.basename(filename))[0]
+        spec = spec_from_file_location(name, filename)
+        config = module_from_spec(spec)
+        spec.loader.exec_module(config)
+        return config
+
+    @staticmethod
+    def _load_profile(profile_path):
+        profile = {}
+        profile_content = ProfileLoader._load_module_from_file(profile_path)
+        for key, value in profile_content.__dict__.items():
+            if not key.startswith("__"):
+                profile[key] = value
+        return profile.copy()
+
+    @staticmethod
+    def load_profile_python(profile_path, parent_profile_path=None):
+        # TODO: implement include() of other profiles? variable substitution?
+        if parent_profile_path:
+            ProfileLoader._load_profile(parent_profile_path)
+        profile = ProfileLoader._load_profile(profile_path)
+        ProfileLoader._init_profile()
+        return profile.copy()
+
+
+class PythonProfilesTest(unittest.TestCase):
+
+    def setUp(self):
+        self.client = TestClient()
+        self.pr1_content = textwrap.dedent("""
+            import os
+            from conans import settings, options, env
+
+            settings.os = "Linux"
+            settings.build_type = "Release"
+
+            options.shared = True
+
+            env.env_var = "bar"
```

Comment:
Lets aim to use the Environment composition defined in https://github.com/conan-io/conan/pull/8534

Reply:
I have added a similar test with append, prepend, delete and so syntax to the `env` object

Reply:
This can wait until #8534 moves forward. You can leave it as is by now, but it would be good to use the same interface, the same that will be used in recipes defined in #8534 in ``Environment``.